File: java-core-plugins/src/main/java/org/unlaxer/tramli/plugins/eventstore/EventLogStorePlugin.java
package org.unlaxer.tramli.plugins.eventstore;

import org.unlaxer.tramli.FlowStore;
import org.unlaxer.tramli.plugins.api.PluginDescriptor;
import org.unlaxer.tramli.plugins.api.StorePlugin;

public final class EventLogStorePlugin implements StorePlugin {
    @Override
    public PluginDescriptor descriptor() {
        return new PluginDescriptor("event-log", "Event Log Store", "Wraps FlowStore with append-only transition and compensation logging.");
    }

    @Override
    public FlowStore wrapStore(FlowStore store) {
        return new EventLogStoreDecorator(store);
    }
}

File: lang/ts-plugins/src/eventstore/event-log-store-plugin.ts
import type { StorePlugin, PluginDescriptor } from '../api/types.js';
import { EventLogStoreDecorator } from './event-log-store-decorator.js';

export class EventLogStorePlugin implements StorePlugin {
  descriptor(): PluginDescriptor {
    return { id: 'eventstore', displayName: 'Event Log Store', description: 'Tenure-lite: append-only transition log with replay' };
  }
  kind() { return 'STORE' as const; }

  wrapStore(store: any): EventLogStoreDecorator {
    return new EventLogStoreDecorator(store);
  }
}
